db/view/view.cc

```diff
@@ -518,18 +518,37 @@ row_marker view_updates::compute_row_marker(const clustering_or_static_row& base
      *      will all unselected columns.
      */
 
-    // WARNING: The code assumes that if multiple regular base columns are present in the view key,
-    // they share liveness information. It's true especially in the only case currently allowed by CQL,
-    // which assumes there's up to one non-pk column in the view key. It's also true in alternator,
-    // which does not carry TTL information.
+    // We need to pick one liveness (timestamp and expiration) for the row
+    // marker, which in general would be a problem if there are multiple
+    // regular base columns in the view key, as those can have different
+    // liveness. This is why we generally (in CQL) do NOT allow multiple
+    // regular base columns in the view key.
+    // We make an exception for Alternator, and allow two columns there (as
+    // the DynamoDB API requires us to allow). We can allow this because
+    // Alternator doesn't use expiration (the "TTL" it does support is
+    // different), and doesn't support user-defined timestamps. But, the
+    // two columns can still have different timestamps - this happens if an
+    // update modifies just one of them. The timestamp of the view update
+    // is the later of the two updated columns.
     const auto& col_ids = base_row.is_clustering_row()
             ? _base_info->base_regular_columns_in_view_pk()
             : _base_info->base_static_columns_in_view_pk();
     if (!col_ids.empty()) {
         auto& def = _base->column_at(base_row.column_kind(), col_ids[0]);
         // Note: multi-cell columns can't be part of the primary key.
         auto cell = base_row.cells().cell_at(col_ids[0]).as_atomic_cell(def);
-        return cell.is_live_and_has_ttl() ? row_marker(cell.timestamp(), cell.ttl(), cell.expiry()) : row_marker(cell.timestamp());
+        auto ts = cell.timestamp();
+        if (col_ids.size() > 1){
+            // As explained above, this case only happens in Alternator,
+            // and we may need to pick a higher ts:
+            auto& second_def = _base->column_at(base_row.column_kind(), col_ids[1]);
+            auto second_cell = base_row.cells().cell_at(col_ids[1]).as_atomic_cell(second_def);
+            auto second_ts = second_cell.timestamp();
```

Comment:
maybe std::max?

Reply:
Almost 40 years ago I learned that the C language has an if() - and I still like it :-)
But I guess that
```c++
if (second_ts > ts) {
   ts = second_ts;
}
```
Is 5 characters longer than 
```c++
ts = std::max(ts, second_ts)
```
So your suggestion has some merit. I'll do it.

To be honest, I'm not sure which of the two is **clearer**, which is usually my goal over few characters.